app/scripts/controllers/main.js

```diff
@@ -39,8 +39,22 @@ angular.module('moodCatApp')
           currentSongService.loadSong(room.song.soundCloudId);
           $log.info("Joining room %s", room.name);
         }
+          room.nextSong = room.song.soundCloudId;
+          this.callNextSong(room);
         return room;
-      }
+      }   
+
+      this.callNextSong = function callNextSong(room){
+
+        if((room.song.duration/1000 - currentSongService.getTime()) < 1){
+          $http.get('/api/rooms/' + room.id).then(function(response) {
+             room.nextSong = response.data.song.soundCloudId;
+             currentSongService.loadSong(room.nextSong);
+             room.song = room.nextSong;
+           });
+        }
+        setTimeout(callNextSong, 1000, room);
```

Comment:
This schedule will break when switching rooms, because the callNextSong for the old room will still be scheduled.